services/MerchStoreService.ts

```diff
@@ -89,9 +95,43 @@ export default class MerchStoreService {
           .merchStoreItem(txn)
           .updateMerchItemsInCollection(uuid, { hidden: changes.archived });
       }
-      if (changes.discountPercentage !== undefined || changes.archived !== undefined) {
+
+      // this part only handles updating the positions of the pictures
+      if (collectionPhotos) {
+        // error on duplicate photo uuids
+        const dupSet = new Set();
+        collectionPhotos.forEach((merchPhoto) => {
+          if (dupSet.has(merchPhoto.uuid)) {
+            throw new UserError(`Multiple edits is made to photo: ${merchPhoto.uuid}`);
+          }
+          dupSet.add(merchPhoto.uuid);
+        });
+
+        const photoUpdatesByUuid = new Map(collectionPhotos.map((merchPhoto) => [merchPhoto.uuid, merchPhoto]));
+
+        currentCollection.collectionPhotos.map((currentPhoto) => {
+          if (!photoUpdatesByUuid.has(currentPhoto.uuid)) return;
+          const photoUpdate = photoUpdatesByUuid.get(currentPhoto.uuid);
+          // photo positions are 0-index
+          return MerchCollectionPhotoModel.merge(currentPhoto, photoUpdate);
+        });
+
+        // make sure the photos are always sorted
```

Comment:
Similar to merchItemPhoto, lets leave the exact indexing of the photos to the frontend. We do not have to validate the 0-indexing and just need to make sure photos positions are updated correctly